controllers/gateway/operator.go

```diff
@@ -76,22 +88,19 @@ func (goc *gwOperationCtx) operate() error {
 		if err != nil {
 			goc.log.Error().Err(err).Msg("failed to create transformer gateway configuration")
 			// mark gateway as failed
-			goc.gw.Status = v1alpha1.NodePhaseError
-			goc.gw, err = gatewayClient.Update(goc.gw)
-			if err != nil {
-				err = goc.reapplyUpdate(gatewayClient)
-				if err != nil {
-					goc.log.Error().Err(err).Msg("failed to update gateway")
-					return err
-				}
-			}
+			goc.markGatewayPhase(v1alpha1.NodePhaseError, "failed to create transformer gateway configuration")
```

Comment:
it's useful to include the `err` in the message.

Reply:
:+1: